.github/workflows/run-e2e-tests.yaml

```diff
@@ -0,0 +1,135 @@
+    outputs:
+      branch: '${{ steps.extract-inputs.outputs.branch }}'
+      delete-bee: '${{ steps.extract-inputs.outputs.delete-bee }}'
+    steps:
+      - name: Get inputs or use defaults
+        id: extract-inputs
+        run: >
+          echo "branch=${{ inputs.branch || 'develop' }}" >> "$GITHUB_OUTPUT"
+          echo "delete-bee=${{ inputs.delete-bee || false }}" >> "$GITHUB_OUTPUT"
+  params-gen:
+    runs-on: ubuntu-latest
+    outputs:
+      project-name: '${{ steps.gen.outputs.project_name }}'
+      bee-name: '${{ env.BEE_NAME }}'
+    steps:
+      - uses: actions/checkout@v4
+      - name: Generate a random billing project name
+        id: gen
+        run: |
+          project_name=$(echo "tmp-billing-project-$(uuidgen)" | cut -c -30)
+          echo "project_name=${project_name}" >> $GITHUB_OUTPUT
+  create-bee-workflow:
+    runs-on: ubuntu-latest
+    permissions:
+      contents: read
+      id-token: write
+    steps:
+      - name: Get actions
+        uses: actions/checkout@v4
+
+      - name: dispatch to terra-github-workflows
+        id: FirstAttemptCreateBee
+        continue-on-error: true
+        uses: ./.github/actions/create-bee
+        with:
+          run_name: "bee-create-${{ env.BEE_NAME }}"
+          bee_name: ${{ env.BEE_NAME }}
+          token: '${{ env.TOKEN }}'
+
+      - name: retry dispatch to terra-github-workflows in case of failure
+        if: steps.FirstAttemptCreateBee.outcome == 'failure'
+        uses: ./.github/actions/create-bee
+        with:
+          run_name: "bee-create-${{ env.BEE_NAME }}-retry"
+          bee_name: ${{ env.BEE_NAME }}
+          token: ${{ env.TOKEN }}
+
+  attach-landing-zone-to-bee-workflow:
+    runs-on: ubuntu-latest
+    needs: [ init-github-context, create-bee-workflow, params-gen ]
+    steps:
+      - name: Get actions
+        uses: actions/checkout@v4
+
+      - name: dispatch to terra-github-workflows
+        id: FirstAttemptLandingZone
+        continue-on-error: true
+        uses: ./.github/actions/landing-zone-attach
+        with:
+          run_name: "${{ env.ATTACH_BP_TO_LZ_RUN_NAME }}"
+          bee_name: ${{ env.BEE_NAME }}
+          token: '${{ env.TOKEN }}'
+          project_name: ${{ needs.params-gen.outputs.project-name }}
+
+      - name: retry dispatch to terra-github-workflows in case of failure
+        if: steps.FirstAttemptLandingZone.outcome == 'failure'
+        uses: ./.github/actions/landing-zone-attach
+        with:
+          run_name: "${{ env.ATTACH_BP_TO_LZ_RUN_NAME }}-retry"
+          bee_name: ${{ env.BEE_NAME }}
+          token: '${{ env.TOKEN }}'
+          project_name: ${{ needs.params-gen.outputs.project-name }}
+
+  run-e2e-test-job:
+    needs:
+      - attach-landing-zone-to-bee-workflow
+      - params-gen
+    permissions:
+      contents: read
+      id-token: write
+    uses: broadinstitute/dsp-reusable-workflows/.github/workflows/run_tsps_e2e_tests.yaml@3a09600bf4ee1580f33ca5cb1ca67f932505436a
```

Comment:
once https://github.com/broadinstitute/dsp-reusable-workflows/pull/29 is merged, this can just point to `main`.  Currently the workflow only exists in our branch